### ISSUE TITLE
[🐛 bug report] Incorrect .obj normals crash render even with `face_normals=False`

### ISSUE THREAD

**Issue by jatentaki** [mitsuba-renderer/mitsuba3#191]
## Summary

Mitsuba3 checks for finiteness of vertex normals and throws an exception in scene loading, even when it would have made no difference due `face_normals=False`. This could be considered working as intended, but it's rather inconvenient for the user.

## System configuration

- Platform: Ubuntu20.04
- Compiler: as on `pip`
- Python version: 3.8.10
- Mitsuba 3 version: 3.0.1
- Compiled variants: (as on `pip`)
    * `scalar_rgb`
    * `scalar_spectral`
    * `llvm_ad_rgb` (used)
    * `cuda_ad_rgb`

## Description

Mitsuba will check for correctness of normals [in .obj loading](https://github.com/mitsuba-renderer/mitsuba3/blob/152352f87b5baea985511b2a80d9f91c3c945a90/src/shapes/obj.cpp#L237) which is separate logic from `face_normals`, so a bad normal (for example `vn 0 0 0`) will break loading. It would be better to ignore thos errors when normals are not used, and ideally I would see some additional flag to just omit bad normals while keeping the remaining ones.

## Reproduction
```python
import mitsuba as mi
mi.set_variant('scalar_rgb')
from mitsuba import ScalarTransform4f as T

# for self-containment let's create the .obj from Python
with open('pyramid.obj', 'w') as obj:
    obj.write('''
v 0.1 0 0
v -0.1 -0.1 0
v -0.1 0.1 0
v 0 0 0.1

vn 1 0 0
vn 0 1 0
vn 0 0 0 # the bad one

f 1//1 2//2 3//3
f 1 2 4
f 1 3 4
f 2 3 4

''')
    
scene = mi.load_dict({
    'type': 'scene',
    'integrator': {'type': 'path'},
    'light': {'type': 'constant'},
    'object': {
        'type': 'obj',
        'filename': 'pyramid.obj',
        'face_normals': False,
        'to_world': T.translate([0, 0, 0]),
        'bsdf': {
            'type': 'diffuse',
            'reflectance': {'type': 'rgb', 'value': [0.4, 0.2, 0.1]},
        },
    },
})
```

**Comment by jatentaki** [comment]
Meanwhile an ugly workaround :)

```python
import tempfile
import numpy as np

class FixObj:
    '''
    mitsuba crashes when loading files with bad normals, therefore this
    creates a temporary file in which bad .obj normals are fixed
    '''
    def __init__(self, path: str):
        self.path = path
        
    def __enter__(self):
        with open(self.path, 'r') as real_obj:
            self.temp = tempfile.NamedTemporaryFile('w')
            
            for line in real_obj:
                if not line.startswith('vn'):
                    self.temp.write(line)
                    continue

                _, x, y, z = line.split()
                x, y, z = float(x), float(y), float(z)
                n = np.array([x, y, z])
                if np.linalg.norm(n) < 1e-5:
                    # replace with a random normal
                    n = np.random.randn(3)
                    n /= np.linalg.norm(n)

                x, y, z = n
                new_line = f'vn {x} {y} {z}\n'

                self.temp.write(new_line)
            
        return self.temp
    
    def __exit__(self, exc_type, exc_value, traceback):
        self.temp.close()

def create_scene(obj_path: str):
    with FixObj(obj_path) as fixed_obj:
        return mi.load_dict({
            'type': 'scene',
            'integrator': {'type': 'path'},
            'light': {'type': 'constant'},
            'object': {
                'type': 'obj',
                'filename': fixed_obj.name,
                'face_normals': False,
                'to_world': T.translate([0, 0, 0]),
                'bsdf': {
                    'type': 'diffuse',
                    'reflectance': {'type': 'rgb', 'value': [0.4, 0.2, 0.1]},
                },
            },
        })
```

**Comment by Speierers** [comment]
Thanks @jatentaki to report this issue. It will be fixed in the next release: [6941c08](https://github.com/mitsuba-renderer/mitsuba3/commit/6941c0834a5be9ba5a208849f96fdfe664f9fa28)
